workflow/controller/operator.go

```diff
@@ -1486,6 +1504,39 @@ func getChildNodeIndex(node *wfv1.NodeStatus, nodes wfv1.Nodes, index int) *wfv1
 	return &lastChildNode
 }
 
+func buildRetryStrategyLocalScope(node *wfv1.NodeStatus, nodes wfv1.Nodes) map[string]string {
+	localScope := make(map[string]string)
+
+	// `retries` variable
+	localScope[common.LocalVarRetries] = strconv.Itoa(len(node.Children) - 1)
+
+	// Default empty value, to be replaced by actual value below if len(node.Children) > 0
+	localScope["retries.last.exitCode"] = "-1"
```

Comment:
This is necessary to pass `expr`'s parser. If `retries.last.exitCode` does not yet exist (as it may during the first retry), an expression such as `{{retries.last.exitCode}} == 1` will evaluate to ` == 1` (or `{{retries.last.exitCode}} == 1` if `allowUnresolved` is set to `true`). In either case, the expression will not be valid and `expr` will error out.

`-1` is a fair default value since it is impossible for a container to exit `-1` and is still an integer such that it will satisfy `expr`.